tests/integration/common-operator-integ-suite.sh

```diff
@@ -14,87 +14,141 @@
-if [ "${OCP}" == "true" ]; then
-  echo "Running on OCP"
-else
-  echo "Running on kind"
-fi
+initialize_variables() {
+  WD=$(dirname "$0")
+  WD=$(cd "$WD" || exit; pwd)
 
-WD=$(dirname "$0")
-WD=$(cd "$WD" || exit; pwd)
+  NAMESPACE="${NAMESPACE:-istio-operator}"
+  DEPLOYMENT_NAME="${DEPLOYMENT_NAME:-istio-operator}"
+  CONTROL_PLANE_NS="${CONTROL_PLANE_NS:-istio-system}"
+  COMMAND="kubectl"
 
-NAMESPACE="${NAMESPACE:-istio-operator}"
-DEPLOYMENT_NAME="${DEPLOYMENT_NAME:-istio-operator}"
-CONTROL_PLANE_NS="${CONTROL_PLANE_NS:-istio-system}"
-COMMAND="kubectl"
+  if [ "${OCP}" == "true" ]; then
+    COMMAND="oc"
+  fi
 
-if [ "${OCP}" == "true" ]; then
-  COMMAND="oc"
-fi
+  echo "Using command: ${COMMAND}"
 
-BRANCH="${BRANCH:-maistra-3.0}"
+  BRANCH="${BRANCH:-maistra-3.0}"
 
-if [ "${OCP}" == "true" ]; then
-  ISTIO_MANIFEST="${WD}/../../config/samples/istio-sample-openshift.yaml"
-else
-  ISTIO_MANIFEST="${WD}/../../config/samples/istio-sample-kubernetes.yaml"
-fi
+  if [ "${OCP}" == "true" ]; then
+    ISTIO_MANIFEST="${WD}/../../config/samples/istio-sample-openshift.yaml"
+  else
+    ISTIO_MANIFEST="${WD}/../../config/samples/istio-sample-kubernetes.yaml"
+  fi
 
-TIMEOUT="3m"
+  TIMEOUT="3m"
 
-check_ready() {
-    local NS=$1
-    local POD_NAME=$2
-    local DEPLOYMENT_NAME=$3
+  # Set a default value for a variable
+  : "${RUN_ONLY_TEST:="false"}"
+}
 
-    echo "Check POD: NAMESPACE: \"${NS}\"   POD NAME: \"${POD_NAME}\""
-    timeout --foreground -v -s SIGHUP -k ${TIMEOUT} ${TIMEOUT} bash --verbose -c \
-    "until $COMMAND  get pod --field-selector=status.phase=Running -n ${NS} | grep ${POD_NAME}; do sleep 5; done"
+get_internal_registry() {
+  # Validate that the internal registry is running, configure the variable to be used in the Makefile. 
+  # If there is no internal registry, we will use the public one defined in Makefile by default
+
+  # Check if the registry pods are running
+  $COMMAND get pods -n openshift-image-registry --no-headers | grep -v "Running" && exit 1
 
-    echo "Check Deployment Available: NAMESPACE: \"${NS}\"   DEPLOYMENT NAME: \"${DEPLOYMENT_NAME}\""
-    $COMMAND  wait deployment "${DEPLOYMENT_NAME}" -n "${NS}" --for condition=Available=True --timeout=${TIMEOUT}
+  # Patch to activate the default route to the registry
+  $COMMAND patch configs.imageregistry.operator.openshift.io/cluster --patch '{"spec":{"defaultRoute":true}}' --type=merge
+
+  # Get the registry route
+  sleep 30 # TODO: improve waiting time to avoid hardcoded sleep
+  oc get routes -n openshift-image-registry
+  URL=$($COMMAND get route default-route -n openshift-image-registry --template='{{ .spec.host }}')
+  # Hub will be equal to the route url/project-name(NameSpace) 
+  export HUB="${URL}/$NAMESPACE"
+  echo "Internal registry URL: ${HUB}"
+
+  # Create namespace where operator will be located
+  # This is needed because the roles are created in the namespace where the operator is deployed
+  oc create namespace "$NAMESPACE" 
```

Comment:
I know that at this point, `oc` is going to be used. But I think we should keep using `${COMMAND}` everywhere. Plus, maybe we should append ` || true` so that the command doesn't fail if the namespace already exists.